### PR TITLE
feat: Error handling middleware (#34)

### PR DESCRIPTION
**Pull request by EvanG13**
### Summary

This PR implements an `error-handling` middleware

### Checklist

- [ ] Tested the code :pray:
- [x] Run Prettier and Eslint `npm run format && npm run lint:fix` 
- [ ] PR title and commit messages are easy for others to follow :eyes:
- [ ] Include a screenshot if it's for a view :camera:

### CODE REVIEW

File: server/middleware/authMiddleware.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+import mongoose from "mongoose";
 
 const authMiddleware = (req: Request, res: Response, next: NextFunction) => {
-	if (req.session.isLoggedIn && req.session.userid) {
+	const userid: string | undefined = req.session.userid;
+	if (
+		req.session.isLoggedIn &&
+		userid &&
+		mongoose.Types.ObjectId.isValid(userid)
+	) {
```
Comment (EvanG13) [inline server/middleware/authMiddleware.ts:10]: updated the auth middleware to now check that the `userid` is a valid id form.

File: server/routes/userEventRouter.ts [hunk 2]
```diff
@@ -7,13 +7,29 @@ import {
...
+	listEvents,
+	errorHandlerMiddleware
+);
+userEventRouter.put(
+	"/:id/update",
+	authMiddleware,
+	update,
+	errorHandlerMiddleware
+);
+userEventRouter.delete(
```
Comment (EvanG13) [inline server/routes/userEventRouter.ts:28]: In addition to adding the `errorHandlerMiddleware` to the routes, I updated the endpoints for `update` and `delete` to require the `device id` as a path param

Comment (CleveyP) [reply server/routes/userEventRouter.ts:28]: what is a device id?

File: server/middleware/authMiddleware.ts (hunk 1 above)
Comment (CleveyP) [reply server/middleware/authMiddleware.ts:10]: appreciate it.

**Review by CleveyP — APPROVED** [review]
lgtm nice job finding out this method of error handling.

**Pull request merged**